projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts

```diff
@@ -932,42 +974,638 @@ export class IgxQueryBuilderTreeComponent implements AfterViewInit, OnDestroy {
         return expression instanceof ExpressionGroupItem;
     }
 
+    /**
+     * @hidden @internal
+     */
+    public onExpressionFocus(expressionItem: ExpressionOperandItem) {
+        if (this.prevFocusedExpression) {
+            this.prevFocusedExpression.focused = false;
+        }
+        expressionItem.focused = true;
+        this.prevFocusedExpression = expressionItem;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public onExpressionBlur(event, expressionItem: ExpressionOperandItem) {
+        if (this._prevFocusedContainer && this._prevFocusedContainer !== event.target.closest('.igx-filter-tree__expression-item')) {
+            expressionItem.focused = false;
+        }
+        this._prevFocusedContainer = event.target.closest('.igx-filter-tree__expression-item');
+    }
+
     /**
      * @hidden @internal
      */
     public onChipRemove(expressionItem: ExpressionItem) {
+        this.exitEditAddMode();
         this.deleteItem(expressionItem);
     }
 
+    /* DRAG AND DROP START*/
+    public sourceExpressionItem: ExpressionItem;
+    public sourceElement: HTMLElement;
+    public targetExpressionItem: ExpressionItem;
+    public targetElement: HTMLElement;
+    public dropUnder: boolean;
+    public dropGhostChipNode: Node;
+    private ghostChipMousemoveSubscription$: Subscription;
```

Comment:
All private props should be prefixed with _.
If we will not move drog&drop into a separate service, all private and public properties should be moved to proper places